### ISSUE TITLE
Do not request "textDocument/documentSymbol" when content changed

### ISSUE THREAD

**Issue by xiong-ang** [TypeFox/monaco-languageclient#209]
Want to implement "Semantic Coloring" by the monaco-languageclient and LSP server.
But I find it does not request "textDocument/documentSymbol".
Desperately want to know whether the function is available? How to open it? Or are there any documents on this?

**Comment by CGNonofr** [comment]
To do so, you'll need a more recent version of monaco editor

I opened a PR (https://github.com/TypeFox/monaco-languageclient/pull/207) to update to monaco 20.0.0

but I didn't implement registerDocumentSemanticTokensProvider and registerDocumentRangeSemanticTokensProvider (they are binded to `unsupported`), so I guess we'll need some changes to make it work

**Comment by xiong-ang** [comment]
Good news. Wish that comes soon!

**Comment by rcjsuen** [comment]
Even if you hook up the semantic tokens provider, you won't get any colouring support because of microsoft/monaco-editor#1833.

**Comment by CGNonofr** [comment]
I just updated my PR to implement the new semantic token API, and I managed to make it work on the docker lsp (https://github.com/rcjsuen/dockerfile-language-server-nodejs) which is the only implementation for the moment

What you need to do is:
- Use the code from my PR
- enable semanticHighlighting:
```typescript
import { StaticServices } from 'monaco-editor/esm/vs/editor/standalone/browser/standaloneServices'
StaticServices.configurationService.get().updateValue('editor.semanticHighlighting', {
  enabled: true
});
```
- implement the `getTokenStyleMetadata` theme method:
```typescript
import { StaticServices } from 'monaco-editor/esm/vs/editor/standalone/browser/standaloneServices'
const ThemeClass = StaticServices.standaloneThemeService.get()._theme.constructor;

interface ITokenStyle {
  readonly foreground?: number;
  readonly bold?: boolean;
  readonly underline?: boolean;
  readonly italic?: boolean;
}

ThemeClass.prototype.getTokenStyleMetadata = function (type: string, modifiers: string[], modelLanguage: string): ITokenStyle {
  // whatever you want based on the type and the modifier
  return {
    foreground: '#ff0000'
  };
}
```
(The last parameter (`modelLanguage`) is undefined for the moment. It's already fixed in vscode but monaco-editor@0.20.0 was bundled before the fix)

- register the feature on  the language client:
```typescript
const client = new MonacoLanguageClient({ ... })
client.registerProposedFeatures()
```
